src/utill.rs

```diff
@@ -189,6 +193,59 @@ pub fn to_hex(bytes: &Vec<u8>) -> String {
     hex_string
 }
 
+pub fn parse_toml(file_path: &PathBuf) -> io::Result<HashMap<String, HashMap<String, String>>> {
+    let file = File::open(file_path)?;
+    let reader = io::BufReader::new(file);
+
+    let mut sections = HashMap::new();
+    let mut current_section = String::new();
+
+    for line in reader.lines() {
+        let line = line?;
+        if line.trim().starts_with('[') {
+            current_section = line
+                .trim()
+                .trim_matches(|p| p == '[' || p == ']')
+                .to_string();
+            sections.insert(current_section.clone(), HashMap::new());
+        } else if line.trim().starts_with('#') {
+            continue;
+        } else if let Some(pos) = line.find('=') {
+            let key = line[..pos].trim().to_string();
+            let value = line[pos + 1..].trim().to_string();
+            if let Some(section) = sections.get_mut(&current_section) {
+                section.insert(key, value);
+            }
+        }
+    }
+
+    Ok(sections)
+}
+
+// Function to parse and log errors for each field
+pub fn parse_field<T: std::str::FromStr>(
+    field_name: &str,
+    value: Option<&String>,
+    default: T,
+) -> T {
+    match value {
+        Some(value) => match value.parse() {
+            Ok(parsed) => parsed,
+            Err(_) => {
+                eprintln!(
+                    "Invalid value or type for {}: '{}', using default.",
+                    field_name, value
+                );
```

Comment:
This is not standard error handling in rust. It's much simpler than this. Make the function return the correct error type in a Result and do `?` instead of `unwraps`.